### DISCUSSION TITLE
HamiltonianPathSolver in Python

### DISCUSSION THREAD

**Discussion by jha-vineet69** [google/or-tools#2374] [Graph and Algorithms]
How can I use HamiltonianPathSolver in Python? Can you guide me to some resources on how to convert this C++ code into python or maybe even how to use this in C++? Like TSP codes examples in the Guides section.

**Comment by Mizux** [comment]
Currently only the Shortest path classes are wrapped in python BUT you should try to search for a pure python implementation of this algorithm instead of using OR-Tools (aka a native implementation i.e. platform dependent package whose the implementation have not been heavily optimized AFAIK).
